paasta_tools/vitesscluster_tools.py

```diff
@@ -72,21 +662,98 @@ def validate(
             return []
 
 
-def load_vitess_instance_config(
+def generate_vitess_instance_config(
+    instance_config: Dict,
+) -> Dict:
+    # Generate the vitess instance config from the yelpsoa config
+
+    cpus = instance_config.get("cpus")
+    mem = instance_config.get("mem")
+    deploy_group = instance_config.get("deploy_group")
+    zk_address = instance_config.get("zk_address")
+    paasta_pool = instance_config.get("paasta_pool")
```

Comment:
e.g., if this was get_pool()`, that would handle there not being a pool set and automatically set this to "default"

(or y'all could override get_pool for VitessDeploymentConfig to return the default pool you'd like to use)